core/server/worker/src/main/java/alluxio/worker/grpc/FileReadHandler.java

```diff
@@ -415,7 +415,8 @@ private void openBlock(BlockReadRequestContext context)
         return;
       }
       BlockReadRequest request = context.getRequest();
-      BlockReader reader = mWorker.createFileReader(request.getOpenUfsBlockOptions().getUfsPath(),
+      BlockReader reader = mWorker.createFileReader(
+          String.valueOf(request.getOpenUfsBlockOptions().getUfsPath().hashCode()),
```

Comment:
why change the file id from a path string to its hashcode? Is the hashcode space large enough to avoid hash collisions?

Reply:
1. The client side uses the hashcode. So, this is to use the same.
2. The original code is purely implemented for Dora. I think this was a typo.  We need an ID here, not a full path file name. This ID is the file ID. 
3. If we use ufspath as ID, this would cause: (a) if the ufs path is s3a://some_bucket/some_file, the final directory would be: /tmp/ramdisk/LOCAL/s3a:/some_bucket/some_file
Please note, the `s3a://` will be split to `s3:/`, and it can not be re-constructed back.
(b) The PagedStoreDir requires the ID to be a long integer. Otherwise, it will report error.

Reply:
>  Is the hashcode space large enough to avoid hash collisions?

This is a BIG technical debt!!
We need to take actions to use 64bit integer or even 128bit integer or even higher number of integer as the FileID.
This must be done as soon as possible.
@beinan

Reply:
>  if the ufs path is s3a://some_bucket/some_file, the final directory would be: /tmp/ramdisk/LOCAL/s3a:/some_bucket/some_file

This is a implementation flaw of `LocalPageStore`. How about using a non-hash codec to transform the path into something that does not resemble a path? e.g. hexadecimal encode that path?

> The PagedStoreDir requires the ID to be a long integer

I think you are referring to `BlockPageId` which was used for blocks that have a numeric ID. It did require that file IDs to be numeric, but this assumption does not hold any more for Dora. Some of the code in `PagedBlockMetaStore` and `PagedBlockStore` needs to be generalized to work with arbitrary file IDs.

Reply:
> 1. The client side uses the hashcode. So, this is to use the same.
> 2. The original code is purely implemented for Dora. I think this was a typo.  We need an ID here, not a full path file name. This ID is the file ID.
> 3. If we use ufspath as ID, this would cause: (a) if the ufs path is s3a://some_bucket/some_file, the final directory would be: /tmp/ramdisk/LOCAL/s3a:/some_bucket/some_file
>    Please note, the `s3a://` will be split to `s3:/`, and it can not be re-constructed back.
>    (b) The PagedStoreDir requires the ID to be a long integer. Otherwise, it will report error.

I tested the codes and found that the third problem mentioned above still exists. It is right to avoid regarding UFS path as file id. There must be somewhere that is still using the UFS path as file id.

Reply:
we may need a common encode function for the purpose of converting a ufs path to a valid path string.

Reply:
> I tested the codes and found that the third problem mentioned above still exists. It is right to avoid regarding UFS path as file id. There must be somewhere that is still using the UFS path as file id.

Can you please test and tell me where the place is.